sdk/tests/deqp/framework/referencerenderer/rrFragmentOperations.js

```diff
@@ -197,16 +212,14 @@ rrFragmentOperations.executeDepthCompare = function(inputFragments, depthFunc, d
                 var depthBufferValue = depthBuffer.getPixDepth(fragSampleNdx, frag.pixelCoord[0], frag.pixelCoord[1]);
                 var sampleDepthFloat = frag.sampleDepths[fragSampleNdx];
 
-                /* convert input float to target buffer format for comparison */
-                var access = new tcuTexture.PixelBufferAccess({
-                    format: depthBuffer.getFormat(),
-                    width: 1,
-                    height: 1,
-                    depth: 1,
-                    data: new ArrayBuffer(8)
-                });
-                access.setPixDepth(sampleDepthFloat, 0, 0, 0);
-                var sampleDepth = access.getPixDepth(0, 0, 0);
+                var sampleDepth;
+                if (convertToDepthBuffer) {
+                    /* convert input float to target buffer format for comparison */
+                    access.setPixDepth(sampleDepthFloat, 0, 0, 0);
+                    sampleDepth = access.getPixDepth(0, 0, 0);
+                } else {
+                    sampleDepth = deMath.clamp(sampleDepthFloat, 0.0, 1.0);
+                }
```

Comment:
There are two macros in this function and this [line](https://android.googlesource.com/platform/external/deqp/+/deqp-dev/framework/referencerenderer/rrFragmentOperations.cpp#212) chooses between the two. Float depth buffer formats use a different macro than the one implemented in #1854, which is what this PR implements.